armory/scenarios/scenario.py

```diff
@@ -85,6 +86,40 @@ def __init__(
             self._set_export_dir(f"{self.export_subdir}_{self.time_stamp}")
         self.results = None
 
+    def user_init(self) -> None:
+        """
+        Import the user-specified initialization module
+            and (optionally) call the specified function name with kwargs
+        """
+        user_init = self.config.get("user_init")
+        if user_init is not None:
+            module = user_init.get("module")
+            log.info(f"Importing user_init module {module}")
+            if not isinstance(module, str):
+                raise ValueError("config: 'user_init' field 'module' must be a str")
+            try:
+                mod = importlib.import_module(module)
+            except ModuleNotFoundError as err:
+                raise ValueError(
+                    f"config: 'user_init' field 'module' '{module}' cannot be imported."
+                    " If using docker, does it need to be added to"
+                    " config['sysconfig']['external_github_repo']?"
+                ) from err
+            name = user_init.get("name")
+            kwargs = user_init.get("kwargs") or {}
+            if name:
+                if kwargs:
+                    kwargs_str = f"**{kwargs}"
+                else:
+                    kwargs_str = ""
+                log.info(f"Calling user_init function {module}.{name}({kwargs_str})")
+                target = getattr(mod, name, None)
+                if not callable(target):
```

Comment:
before we check if it's callable, should there be another check that it's not None?

If you specify a name that doesnt actually exist, you'll get 
```
ValueError <module.function_that_doesnt_exist> is not callable
```

rather than an error indicating the name can't even be found